command/agent/agent.go

```diff
@@ -222,6 +222,22 @@ func (a *Agent) setupClient() error {
 	conf.Node.Name = a.config.NodeName
 	conf.Node.Meta = a.config.Client.Meta
 	conf.Node.NodeClass = a.config.Client.NodeClass
+	httpAddr := a.config.BindAddr
+	if a.config.Addresses.HTTP != "" {
```

Comment:
@dadgar Don't quite get it. We are trying to find out the right thing to advertise. So if bind addr, addresses.HTTP and AdverticeAddrs.HTTP, all three are present we pick up the advertice addr, but if no advertise addr is present but addresses.http and bind addr is present we choose addresses.http.